### PR TITLE
makes markers tappable, campuses tappable, and always show current campus first

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

This PR handles a slew of issues / tweaks that we discovered as valuable during the QA process (and were mentioned by both willow and christ fellowship) 

The enhancements are: 

1. The campus cards themselves are now tappable
2. Tapping on a marker will change your current "focused" campus to that marker
3. If you have a current selected campus, it will always be the first campus in the list (and the one that you see when you open the mapview) 

### What design trade-offs/decisions were made?

This component should eventually move into its own package.

### How do I test this PR?

1. Open the Location Finder. 
2. Tap on a campus marker that's far from you (for example, Dallas for those of us in Cincy)  
3. Tap on the Campus Card (not the "Choose Campus" button)
4. Open Location Finder again. The campus you selected should be the first campus in the list.

### What automated tests did you write?

n/a

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by vinnyjth** [comment]
|  |  |  |
|-----------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------|
|                                                                                                                             |                                                                                                                             |                                                                                                                             |

File: packages/apolloschurchapp/src/user-settings/Locations/index.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { Query, Mutation } from 'react-apollo';
 import { Dimensions } from 'react-native';
 import Geolocation from 'react-native-geolocation-service';
 import { PaddedView, ButtonLink } from '@apollosproject/ui-kit';
+import { get } from 'lodash';
```
Comment (jordanwade) [inline packages/apolloschurchapp/src/user-settings/Locations/index.js]: Does the Apollos project have treeshaking setup?

Comment (burkeshartsis) [reply packages/apolloschurchapp/src/user-settings/Locations/index.js]: I do not believe the default packager (metro) supports tree shaking. However, recent updates and changes to React Native have helped pave the way to a future full of shook trees. In RN 59 they announced inline requires which would give you manual control to perform some of the tasks tree shaking does.

File: packages/apolloschurchapp/src/user-settings/Locations/Marker.js [hunk 2]
```diff
@@ -25,13 +25,15 @@ const MarkerRingView = styled(({ theme }) => ({
...
-const StyledMarker = memo(({ latitude, longitude, opacityStyle }) => (
-  <Marker coordinate={{ latitude, longitude }}>
-    <Animated.View style={opacityStyle}>
-      <MarkerRingView>
-        <MarkerView />
-      </MarkerRingView>
-    </Animated.View>
+const StyledMarker = memo(({ latitude, longitude, opacityStyle, onPress }) => (
+  <Marker onPress={onPress} coordinate={{ latitude, longitude }}>
+    <Touchable>
```
Comment (erikkgagewhitt) [inline packages/apolloschurchapp/src/user-settings/Locations/Marker.js]: I'm curious, if the Marker component itself supports `onPress`, is the touchable component necessary?

Comment (vinnyjth) [reply packages/apolloschurchapp/src/user-settings/Locations/Marker.js]: Oh! Good catch @erikkgagewhitt :D I need to remove that touchable, it's actually not needed. The Marker `onPress` is handling all the interaction (the `Touchable` is dead code)

**Comment by erikkgagewhitt** [comment]
Does this appear in the core app? If not, is there a way to run the Willow Creek app?

**Pull request merged**